diffkemp/simpll/passes/VarValueDependencySlicer.cpp

```diff
@@ -0,0 +1,123 @@
+#include <llvm/IR/Constant.h>
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/GlobalValue.h>
+#include <llvm/IR/GlobalVariable.h>
+#include <llvm/IR/InstrTypes.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/raw_ostream.h>
+#include <utility>
+
+using namespace llvm;
+
+PreservedAnalyses VarValueDependencySlicer::run(Module &Mod,
+                                                ModuleAnalysisManager &mam,
+                                                GlobalVariable *Var,
+                                                std::string VarValue) {
+    Constant *initializer, *newConst;
+    SmallVector<Instruction *, 128> *listOfLoadInstructions =
+            new SmallVector<Instruction *, 128>();
+    SmallVector<std::pair<Instruction *, int>, 128>
+            *listOfLoadInstructionUsers =
+                    new SmallVector<std::pair<Instruction *, int>, 128>();
+
+    if (Var->hasInitializer())
+        initializer = Var->getInitializer();
+    else
+        return PreservedAnalyses::all();
+
+    if (VarValue != "default") {
+        newConst = getConstantFromString(initializer, VarValue);
+
+        if (newConst == NULL)
+            return PreservedAnalyses::none();
+    } else
+        newConst = initializer;
+
+    for (User *user : Var->users()) {
+        if (LoadInst *loadInst = dyn_cast<LoadInst>(user)) {
+            listOfLoadInstructions->push_back(loadInst);
+
+            for (User *instrUser : loadInst->users()) {
+                if (Instruction *instrOfGv = dyn_cast<Instruction>(instrUser)) {
+                    int index = GetIndexOfGlobalVariableOperand(instrOfGv,
+                                                                loadInst);
+                    listOfLoadInstructionUsers->push_back(
+                            std::make_pair(instrOfGv, index));
+                }
+            }
+        }
+    }
+
+    while (!listOfLoadInstructionUsers->empty()) {
+        std::pair<Instruction *, int> pair =
+                listOfLoadInstructionUsers->pop_back_val();
+        Instruction *instruction = pair.first;
+        int index = pair.second;
+
+        instruction->setOperand(index, newConst);
+    }
+
+    while (!listOfLoadInstructions->empty()) {
+        Instruction *instruction = listOfLoadInstructions->pop_back_val();
+        instruction->eraseFromParent();
+    }
+
+    return PreservedAnalyses::all();
+}
+
+Constant *VarValueDependencySlicer::getConstantFromString(Constant *initializer,
+                                                          std::string Value) {
+    if (const ConstantInt *CI = dyn_cast<ConstantInt>(initializer)) {
+        uint64_t val;
+        int valtmp;
+        try {
+            valtmp = std::stoi(Value);
+        } catch (const std::invalid_argument &ia) {
+            exit(10);
+        }
+        val = valtmp;
+        ConstantInt *CI2 = ConstantInt::get(CI->getType(), val, true);
+
+        return CI2;
```

Comment:
Directly return `ConstantInt::get(CI->getType(), val, true)`, there's no need to use a temporary variable.